pkg/controllers/gateway/gateway_controller.go

```diff
@@ -269,8 +278,20 @@ func (r *GatewayReconciler) reconcileDownstreamFromUpstreamGateway(ctx context.C
 		}
 	}
 
+	longestTTL := 0
```

Comment:
this seems really odd here? should it not be a part of the DNS service perhaps? Also wondering if we do this do we really need the TTL on the dns record.

Reply:
Also I am thinking that what we really want is the TTL for the record for the gateway we are about to remove. IE the A or CNAME record that is actually pointing at the gateway instance / lb instance.
For example we could have a CNAME lb.my.zone.com that is a CNAME to ie.lb.my.zone.com that has several weighted records 1 per gateway instance that are A records with IPs. The A records will likely have a lower TTL while the geo CNAMEs will likely have a longer TTL. Anyway point is we want the TTL for the record that is specific to the instance we are removing.

Reply:
So if TTL is in the DNSPolicy it is the TTL  for the record pointing at the gateway record and we use it to then calculate what "our" Geo / Weighted record values should be. cc @mikenairn 
If we do this then the TTL in the DNSPolicy is all we need

Reply:
after more conversation. We will remove TTL from the policy as it is likely to add confusion and it is not clear what the usecases are we would be solving. We will also not expose the grace period but rather set that internally for now